pkg/cloud/scope/cluster.go

```diff
@@ -98,6 +101,28 @@ type ClusterScope struct {
 	controllerName  string
 }
 
+// KubeadmControlPlaneBindPort returns the bind port used in KCP kubeconfig.
+func (s *ClusterScope) KubeadmControlPlaneBindPort() (int32, error) {
+	if s.Cluster.Spec.ControlPlaneRef != nil {
+		cpObjectKey := client.ObjectKey{
+			Namespace: s.Cluster.Namespace,
+			Name:      s.Cluster.Spec.ControlPlaneRef.Name,
+		}
+		var kubeadmCP controlplanev1.KubeadmControlPlane
+		err := s.client.Get(context.TODO(), cpObjectKey, &kubeadmCP)
+
+		if err != nil && !apierrors.IsNotFound(err) {
```

Comment:
```
if err := s.client.Get(context.TODO(), cpObjectKey, &kubeadmCP); err !=nil && !apierrors.IsNotFound(err){
return 0, err
}
```

Reply:
IMO the current one is more readable as otherwise it will become a longer statement.

Reply:
> Can use reflect.DeepEqual(a,b)

Since this is just a nil check and not any complex structure where we have to match each and every property of the struct with each other, i think it should be fine, wdyt?